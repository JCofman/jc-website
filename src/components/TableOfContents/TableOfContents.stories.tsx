import React from 'react';
import { storiesOf } from '@storybook/react';
import TableOfContents from './TableOfContents';

const items = [
  {
    url: `#exciting-stuff`,
    title: `Exciting Stuff`,
  },
  {
    url: `#what-i-have-learned-in-a-nutshell`,
    title: `What I have learned in a nutshell`,
  },
  {
    url: `#some-improvement-ideas`,
    title: `Some improvement ideas`,
  },
];
storiesOf(`TableOfContents`, module).add(`Basic`, () => {
  return <TableOfContents items={items} />;
});
