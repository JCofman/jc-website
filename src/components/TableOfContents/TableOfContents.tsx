import React from 'react';

import styled from 'styled-components';
import PropTypes from 'prop-types';
import { FaListUl } from 'react-icons/fa';

import Divider from '../Divider';
import Heading from '../Heading';

const StyledTableOfContentsNavigation = styled.nav`
  display: none;
  padding: 1.6rem;
  margin-left: 1rem;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;

  max-width: 300px;
  font-family: 'Raleway', sans-serif;
  color: var(--color-text);
  background: var(--color-background);
  position: sticky;
  top: 70px;

  ${(props) => props.theme.xlarge} {
    display: flex;
  }
  h2 {
    font-size: 1.8rem;
    font-weight: var(--font-weight-medium);
    text-transform: uppercase;
    letter-spacing: 0.2rem;
    text-rendering: optimizelegibility;
    margin-bottom: 16px;
  }

  a {
    font-size: var(--font-size-sm);
    display: block;
    opacity: 0.8;
    text-decoration: none;
    transition: opacity 500ms ease 0s;
    margin-top: var(--margin-1);
    text-align: left;
  }

  a:hover,
  a:focus {
    color: var(--color-primary);
  }
`;

const TableOfContents = (props) => {
  const { items } = props;
  return (
    <div>
      {items && items.length > 0 ? (
        <StyledTableOfContentsNavigation>
          <Heading appearance={`H2`}>
            <FaListUl></FaListUl> Table of contents
          </Heading>

          <Divider></Divider>
          {items.map((item) => {
            const { url, title } = item;
            return (
              <a key={title} href={`${url}`}>
                {title}
              </a>
            );
          })}
        </StyledTableOfContentsNavigation>
      ) : null}
    </div>
  );
};

PropTypes.displayName = `TableOfContents`;

export default TableOfContents;
