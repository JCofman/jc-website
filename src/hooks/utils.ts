const throttle = (func, threshold = 250, scope) => {
  let last, deferTimer;
  return function () {
    let context = scope || this;

    let now = Date.now(),
      args = arguments;
    if (last && now < last + threshold) {
      // hold on to it
      clearTimeout(deferTimer);
      deferTimer = setTimeout(function () {
        last = now;
        func.apply(context, args);
      }, threshold);
    } else {
      last = now;
      func.apply(context, args);
    }
  };
};

export { throttle };
