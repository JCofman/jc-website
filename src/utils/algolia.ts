// const postQuery = `{
//     posts: allMarkdownRemark(
//       filter: { fileAbsolutePath: { regex: "/posts/" } }
//     ) {
//       edges {
//         node {
//           objectID: id
//           frontmatter {
//             title
//             path
//             date(formatString: "MMM D, YYYY")
//             tags
//           }
//           excerpt(pruneLength: 5000)
//         }
//       }
//     }
//   }`;
// const flatten = arr =>
//   arr.map(({ node: { frontmatter, ...rest } }) => ({
//     ...frontmatter,
//     ...rest,
//   }));
// const settings = { attributesToSnippet: [`excerpt:20`] };
// const queries = [
//   {
//     query: pageQuery,
//     transformer: ({ data }) => flatten(data.pages.edges),
//     indexName: `Pages`,
//     settings,
//   },
//   {
//     query: postQuery,
//     transformer: ({ data }) => flatten(data.posts.edges),
//     indexName: `Posts`,
//     settings,
//   },
// ];
// module.exports = queries;
