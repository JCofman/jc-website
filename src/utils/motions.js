const iconMotion = {
  rest: {
    rotate: 0,
  },
  hover: {
    rotate: 9,
    transition: { type: 'spring', stiffness: 900 },
  },
};

export { iconMotion };
