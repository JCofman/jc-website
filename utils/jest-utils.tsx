import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { themes, breakPoints, maxWidth, themeTransition, colors } from '../src/components/Layout/Theme';

export const renderWithTheme = (component) =>
  render(
    <ThemeProvider
      theme={{
        mode: themes[`light`],
        ...breakPoints,
        maxWidth,
        themeTransition,
        colors: { ...colors },
      }}
    >
      {component}
    </ThemeProvider>
  );
