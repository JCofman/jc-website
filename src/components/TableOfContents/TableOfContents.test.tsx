import React from 'react';
import { renderWithTheme } from '../../../utils/jest-utils';

import TableOfContents from './TableOfContents';

const items = [
  {
    url: `#exciting-stuff`,
    title: `Exciting Stuff`,
  },
  {
    url: `#what-i-have-learned-in-a-nutshell`,
    title: `What I have learned in a nutshell`,
  },
  {
    url: `#some-improvement-ideas`,
    title: `Some improvement ideas`,
  },
];

test(`Divider is rendered properly`, () => {
  const { container } = renderWithTheme(<TableOfContents items={items} />);

  expect(container).toMatchSnapshot();
});

test.todo(
  `TableOfContents can render multiple items`

  // () => {
  //   const { getByRole } = renderWithTheme(<TableOfContents items={items} />);
  //   getByRole(`link`, { name: /some improvement ideas/i });
  //   getByRole(`link`, { name: /what I have learned in a nutshell/i });
  //   getByRole(`link`, { name: /exciting Stuff/i });
  // }
);

test(`TableOfContents can render no items`, () => {
  const { queryByRole } = renderWithTheme(<TableOfContents items={[]} />);
  expect(queryByRole(`link`)).not.toBeInTheDocument();
});
