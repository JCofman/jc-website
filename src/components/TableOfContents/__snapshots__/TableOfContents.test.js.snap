// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider is rendered properly 1`] = `
.c3 {
  fill: var(--color-text);
}

.c2 {
  display: block;
  width: 55px;
  height: 12px;
  margin: 0 auto;
}

.c1 {
  font-family: 'Raleway',sans-serif;
  -webkit-transform: skew(-5deg) rotate(-1deg);
  -ms-transform: skew(-5deg) rotate(-1deg);
  transform: skew(-5deg) rotate(-1deg);
  margin-bottom: 2rem;
  color: var(--color-text);
  text-shadow: var(--text-shadow);
  font-size: var(--font-size-5xl);
}

.c1 a {
  color: var(--color-text);
  -webkit-transition: color var(--theme-transition);
  transition: color var(--theme-transition);
}

.c1 a:hover,
.c1 a:focus {
  color: var(--color-primary);
  -webkit-transition: none;
  transition: none;
}

.c1:before {
  width: 0;
  height: 0;
  opacity: 0.2;
  border-left: 1em solid transparent;
  border-right: 1em solid transparent;
  border-bottom: 2em solid var(--color-text);
  content: '';
  pointer-events: none;
  position: absolute;
  z-index: -1;
  -webkit-transform: translateX(-0.5em) translateY(-1.5rem);
  -ms-transform: translateX(-0.5em) translateY(-1.5rem);
  transform: translateX(-0.5em) translateY(-1.5rem);
}

.c0 {
  display: none;
  padding: 1.6rem;
  margin-left: 1rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  max-width: 300px;
  font-family: 'Raleway',sans-serif;
  color: var(--color-text);
  background: var(--color-background);
  position: -webkit-sticky;
  position: sticky;
  top: 70px;
}

.c0 h2 {
  font-size: 1.8rem;
  font-weight: var(--font-weight-medium);
  text-transform: uppercase;
  -webkit-letter-spacing: 0.2rem;
  -moz-letter-spacing: 0.2rem;
  -ms-letter-spacing: 0.2rem;
  letter-spacing: 0.2rem;
  text-rendering: optimizelegibility;
  margin-bottom: 16px;
}

.c0 a {
  font-size: var(--font-size-sm);
  display: block;
  opacity: 0.8;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: opacity 500ms ease 0s;
  transition: opacity 500ms ease 0s;
  margin-top: var(--margin-1);
  text-align: left;
}

.c0 a:hover,
.c0 a:focus {
  color: var(--color-primary);
}

@media (min-width:1200px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div>
  <div>
    <nav
      class="c0"
    >
      <h2
        class="c1"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M48 48a48 48 0 1 0 48 48 48 48 0 0 0-48-48zm0 160a48 48 0 1 0 48 48 48 48 0 0 0-48-48zm0 160a48 48 0 1 0 48 48 48 48 0 0 0-48-48zm448 16H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
          />
        </svg>
         Table of contents
      </h2>
      <svg
        class="c2"
      >
        <g
          fill="none"
          fill-rule="evenodd"
          stroke="none"
          stroke-width="1"
        >
          <g
            class="c3"
            transform="translate(-483.000000, -1059.000000)"
          >
            <g
              transform="translate(182.000000, 936.000000)"
            >
              <g
                transform="translate(301.000000, 123.000000)"
              >
                <rect
                  fill-opacity="0.3"
                  height="12"
                  width="35"
                  x="0"
                  y="0"
                />
                <rect
                  fill-opacity="0.1"
                  height="12"
                  width="35"
                  x="20"
                  y="0"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
      <a
        href="#exciting-stuff"
      >
        Exciting Stuff
      </a>
      <a
        href="#what-i-have-learned-in-a-nutshell"
      >
        What I have learned in a nutshell
      </a>
      <a
        href="#some-improvement-ideas"
      >
        Some improvement ideas
      </a>
    </nav>
  </div>
</div>
`;
